src/models/d2Model.ts

```diff
@@ -88,6 +92,9 @@ export abstract class D2Model {
 
 export class OrganisationUnitModel extends D2Model {
     protected static metadataType = "organisationUnit";
+    protected static groupFilterName = "organisationUnitGroups";
+    protected static levelFilterName = "level";
```

Comment:
Just a comment, if you are convinced it's better (or simpler) this way, go ahead: isn't a bit odd to have references in the abstract class to filter fields not present in all models? (groups, levels, more that we could potentially have).

The most simple alternative I see: a method on the derived classes that is called (if defined) from listMethod with the current object filter and returns an array of d2 filters.

Reply:
I thought of it too, groups makes sense since DHIS2 has multiple metadata grouped and all of them uses the same approach. Here we're re-using the same logic for Indicators, ValidationRules and OrganisationUnits but if we expanded the reusability would be even more.

With levels I agree that it should probable be moved to list custom filters, there're only 2 metadata classes that allow level filtering: OrganisationUnits and DataApproval.

Even though I left it inside the generic because both of them use the same filtering approach but I should probable move it outside.